src/devtools/client/debugger/src/components/Editor/LineNumberTooltip.tsx

```diff
@@ -71,6 +83,92 @@ function Wrapper({
   return <div className="static-tooltip-content bg-gray-700">{children}</div>;
 }
 
+function runAnalysisOnLine(line: number): UIThunkAction {
+  return async (dispatch, getState, { ThreadFront }) => {
+    // A lot of this logic is reused from logpoint.ts
+    // It is possible to instead repurpose that function here, but increasingly
+    // I want logpoint.ts to be... for logpoints. Part of what has made our
+    // analysis code such a mess is trying to repurpose an analysis runner to
+    // work for many different devtools specific workflows. It would be better
+    // if many components used the primitives from `protocol` instead. If it
+    // turns out that those primitives are getting used the same way all the
+    // time, maybe it's time to add some new things to the `protocol` folder.
```

Comment:
I don't disagree with the overall sentiment (that DRY isn't always a good choice) but I hope our long-term goal is for our _components_ not to have this kind of logic in them at all.